### ISSUE TITLE
Investigate potential memory leaks

### ISSUE THREAD

**Issue by pirog** [kalabox/kalabox#1368]
Bug Report
==========

See the following
#1252 
#1323 
#1340 

**For those experiencing performance issues w/Kalabox:**

We've identified the GUI as the instigator of these performance issues. A temporary workaround is to use the CLI exclusively. Running "kbox up" and then simply typing "kbox" to get a list of commands in your terminal can get you started.

**Comment by reynoldsalec** [comment]
First screenshot: After Kalabox has been started, starting up an app:

![screen shot 2016-05-30 at 1 45 14 pm](https://cloud.githubusercontent.com/assets/1153738/15658079/e4d85ea0-266c-11e6-9a57-ecc5ac093e0a.png)

Second screenshot: one app running, but otherwise completely idle:

![screen shot 2016-05-30 at 1 47 03 pm](https://cloud.githubusercontent.com/assets/1153738/15658093/1596c888-266d-11e6-914d-4b98aa161272.png)


I've taken several of these and all seem to show the same upwards trend; even between recording separate events sequentially, it doesn't seem that at any point the memory usage cycles down, I only see higher numbers.

Looking at the stack traces provided, I'm seeing code related to our recloop.js service in the guiEngine show up consistently, which makes sense, since it appears to be a recursive loop function to keep certain tasks running throughout execution.

This would also make sense with reports that performance issues really became impactful in the upgrade to Alpha4, since recloop.js was last changed during the Alpha4 development cycle.

I'm going to dig deeper into those changes and see if this is a good place to get traction.

**Comment by reynoldsalec** [comment]
Ok, so recloop.js is used for three things:

- Polling the CLI for changes in site state
- Polling the CLI for any errors that may have appeared
- Monitoring the list of errors to bring up the error modal when appropriate

The last use can be replaced by a watch statement in Angular. I'm less certain what to do about the first two, but guessing we'll need to fix the functionality of recloop.

**Comment by ikorgik** [comment]
I have VBoxHeadless CPU [#1340](https://github.com/kalabox/kalabox/issues/1340#issuecomment-221806177) issue too.

**Comment by reynoldsalec** [comment]
Thanks @ikorgik!

Update on this issue:

After some testing, it does appear that we've found (and have a fix for) the root reason behind memory leaks revolving around the 'nwjs helper' app as described in #1323. We'll deploy that in our next release, once we've resolved these issues:

- Tests failing on memory leak branch right now
- New errors in console

Regarding #1340 and the syncthing process, these may be much more difficult to address. We need to create some documentation around syncthing, why we've chosen it as a solution, and what we see as the project's long-term roadmap for dealing with file system sharing in general.
